File: Python/50-数组中重复的数字.py
# -*- coding:utf-8 -*-
class Solution:
    # 这里要特别注意~找到任意重复的一个值并赋值到duplication[0]
    # 函数返回True/False
    # 2 3 1 0 2 5 3
    def duplicate(self, numbers, duplication):
        # write code here
        n = len(numbers)
        for i in range(0, n):
            cur = numbers[i]
            if cur == i:
                continue
            if cur == numbers[cur]:
                duplication[0] = cur
                return True
            else:
                numbers[cur], numbers[i] = numbers[i], numbers[cur]
        return False

File: C++/50-数组中重复的数字.cpp
class Solution
{
public:
    // Parameters:
    //        numbers:     an array of integers
    //        length:      the length of array numbers
    //        duplication: (Output) the duplicated number in the array number
    // Return value:       true if the input is valid, and there are some duplications in the array number
    //                     otherwise false
    bool duplicate(int numbers[], int length, int *duplication)
    {
        for (int i = 0; i < length; i++)
        {
            int cur = numbers[i];
            if(cur == i)
                continue;
            if(cur == numbers[cur]){
                *duplication = cur;
                return true;
            }
            else{
                int temp = numbers[cur];
                numbers[cur] = numbers[i];
                numbers[i] = temp;
            }
        }
        return false;
    }
};